### DISCUSSION TITLE
Awkward syntax highlighting with additional_vim_regex_highlighting=false

### DISCUSSION THREAD

**Discussion by kirillbobyrev** [nvim-treesitter/nvim-treesitter#1951] [Q&A]
Hi! I've just started using nvim-TreeSitter and it's awesome!

However, I have a problem: when I set `additional_vim_regex_highlighting = false` (which is said to slow down the editor), the syntax highlighting becomes really awkward (see the examples attached).

# Before (normal)

<img width="664" alt="Screenshot 2021-10-29 at 16 25 13" src="https://user-images.githubusercontent.com/3352968/139452814-be0a2054-334c-48e5-b078-93599fbf4916.png">

# After

<img width="660" alt="Screenshot 2021-10-29 at 16 25 33" src="https://user-images.githubusercontent.com/3352968/139452820-eb7f11b8-da9d-4aeb-a01a-7915550a0a19.png">

When I do `:syntax on` in my editor becomes normal (picture one). However, when I want to put `syntax on` into my `.vimrc` it breaks the highlighting and everything looks like in the second picture.

To sum up:

- When I have `additional_vim_regex_highlighting = true`, regardless of whether I have `syntax on` in my config the highlighting looks great (the first picture).
- With `additional_vim_regex_highlighting = false` I have to **manually** do `:syntax on` when the editor loads in order to make the highlighting to work, otherwise it looks like the second picture.

**Comment by stsewd** [comment]
Hi, the vim parser is still in progress, can you share your `.vimrc`? Also, some highlights may be because of your colorscheme.

**Reply by kirillbobyrev** [reply]
Thank you for looking into this! Here is my [config](https://github.com/kirillbobyrev/dotfiles/tree/main/.config/nvim): interesting bits are [init.vim](https://github.com/kirillbobyrev/dotfiles/blob/main/.config/nvim/init.vim) and [plugins.vim](https://github.com/kirillbobyrev/dotfiles/blob/main/.config/nvim/plugins.vim).

Unfortunately, it happens regardless of the filetype, here are the pictures from the supported language (C++). Example:

## When I just open the file

<img width="657" alt="Screenshot 2021-11-06 at 08 59 14" src="https://user-images.githubusercontent.com/3352968/140602644-d073bfef-5d0d-4d1d-986f-449a7d85e626.png">

## After _manually_ doing `:syntax on`

<img width="657" alt="Screenshot 2021-11-06 at 08 59 19" src="https://user-images.githubusercontent.com/3352968/140602647-61b090d6-9896-43a4-9b3d-d91af06fdc08.png">

**Comment by marcelmanz** [comment]
Any solution for this?? What happens to me is that with `additional_vim_regex_highlighting = false` tree sitter for `diff` and `gitcommit` does not work 🤷🏽
